lib/internal/test_runner/test.js

```diff
@@ -94,72 +94,59 @@ function stopTest(timeout, signal) {
   });
 }
 
-class TestContext {
-  #test;
-
-  constructor(test) {
-    this.#test = test;
+class BaseContext {
+  constructor(context) {
+    this._context = context;
```

Comment:
Let's not use an underscore leading property, instead we can use a `Symbol`.

Reply:
I disagree. We should continue using private properties like we were before.

Reply:
@cjihrig I think inheritance is also important to reduce duplicated code. But unfortunately JS doesn't support `protected` keyword for now. So there is no solution to protect member variable for this case because `child class` uses `parent class`'s member variable. Do you mean that we don't use inheritance only for this case?

Reply:
I would prefer keeping this private. To be honest, I don't see a lot of benefit to the changes in this PR because there is very little duplication. On the other hand, using a Symbol or otherwise userland accessible property gives access to the internals of the test runner that I have been trying to prevent.

Reply:
Thank you for feedback. I'll close this PR soon.